src/test/mediapipe/ovms_kfs_calculator.cc

```diff
@@ -0,0 +1,116 @@
+    {                                                                                               \
+        auto* err = C_API_CALL;                                                                     \
+        if (err != nullptr) {                                                                       \
+            uint32_t code = 0;                                                                      \
+            const char* msg = nullptr;                                                              \
+            OVMS_StatusGetCode(err, &code);                                                         \
+            OVMS_StatusGetDetails(err, &msg);                                                       \
+            LOG(ERROR) << "Error encountred in OVMSKFSPassCalculator:" << msg << " code: " << code; \
+            OVMS_StatusDelete(err);                                                                 \
+            RET_CHECK(err == nullptr);                                                              \
+        }                                                                                           \
+    }
+#define CREATE_GUARD(GUARD_NAME, CAPI_TYPE, CAPI_PTR) \
+    std::unique_ptr<CAPI_TYPE, decltype(&(CAPI_TYPE##Delete))> GUARD_NAME(CAPI_PTR, &(CAPI_TYPE##Delete));
+}  // namespace
+
+class OVMSKFSPassCalculator : public CalculatorBase {
+    OVMS_Server* cserver{nullptr};
+    OVMS_ServerSettings* _serverSettings{nullptr};
+    OVMS_ModelsSettings* _modelsSettings{nullptr};
+    std::unordered_map<std::string, std::string> outputNameToTag;
+    KFSResponse* response;
+
+public:
+    static absl::Status GetContract(CalculatorContract* cc) {
+        RET_CHECK(!cc->Inputs().GetTags().empty());
+        RET_CHECK(!cc->Outputs().GetTags().empty());
+        cc->Inputs().Tag("REQUEST").Set<const KFSRequest*>();
+        cc->Outputs().Tag("RESPONSE").Set<KFSResponse*>();
+
+        return absl::OkStatus();
+    }
+
+    absl::Status Close(CalculatorContext* cc) final {
+        delete response;
+        return absl::OkStatus();
+    }
+    absl::Status Open(CalculatorContext* cc) final {
+        for (CollectionItemId id = cc->Inputs().BeginId(); id < cc->Inputs().EndId(); ++id) {
+            if (!cc->Inputs().Get(id).Header().IsEmpty()) {
+                cc->Outputs().Get(id).SetHeader(cc->Inputs().Get(id).Header());
+            }
+        }
+        if (cc->OutputSidePackets().NumEntries() != 0) {
+            for (CollectionItemId id = cc->InputSidePackets().BeginId(); id < cc->InputSidePackets().EndId(); ++id) {
+                cc->OutputSidePackets().Get(id).Set(cc->InputSidePackets().Get(id));
+            }
+        }
+        cc->SetOffset(TimestampDiff(0));
+        OVMS_ServerNew(&cserver);
+
+        return absl::OkStatus();
+    }
+
+    absl::Status Process(CalculatorContext* cc) final {
+        const KFSRequest* request = cc->Inputs().Tag("REQUEST").Get<const KFSRequest*>();
+        response = new KFSResponse();
```

Comment:
We show here that we create new ptr. When it is cleaned-up (deleted)?

Reply:
In absl::Status Close(CalculatorContext* cc) final {

Reply:
Changed to shared_ptr now.